org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/AbstractDeclarativeIdeQuickfixProvider.java

```diff
@@ -0,0 +1,86 @@
+package org.eclipse.xtext.ide.editor.quickfix;
+/*******************************************************************************
+ * Copyright (c) 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.xtext.ide.server.codeActions.ICodeActionService2.Options;
+
+import com.google.common.base.Predicate;
+import com.google.common.base.Strings;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+
+public class AbstractDeclarativeIdeQuickfixProvider implements IQuickFixProvider {
+
+	private static final Logger LOG = Logger.getLogger(AbstractDeclarativeIdeQuickfixProvider.class);
+
+	@Inject
+	private Provider<DiagnosticResolutionAcceptor> issueResolutionAcceptorProvider;
+
+	private Predicate<Method> getFixMethodPredicate(final String issueCode) {
```

Comment:
No need for final arguments or final parameters, they don't offer much value and make the code harder to read. There's quite a few more in the code